iceoryx_posh/include/iceoryx_posh/popo/rpc_header.hpp

```diff
@@ -0,0 +1,160 @@
+    static constexpr uint8_t RPC_HEADER_VERSION{1U};
+
+    static constexpr uint32_t UNKNOWN_CLIENT_QUEUE_INDEX{std::numeric_limits<uint32_t>::max()};
+
+    /// @brief The RpcBaseHeader version is used to detect incompatibilities for record&replay functionality
+    /// @return the RpcBaseHeader version
+    uint8_t getRpcHeaderVersion() const;
+
+    /// @briet Obtains the sequence ID of the RPC message
+    /// @return the sequenceId of the RPC message
+    int64_t getSequenceId() const noexcept;
+
+    /// @brief Get the pointer to the ChunkHeader
+    /// @return the pointer to the ChunkHeader
+    mepoo::ChunkHeader* getChunkHeader() noexcept;
+
+    /// @brief Get the const pointer to the ChunkHeader
+    /// @return the const pointer to the ChunkHeader
+    const mepoo::ChunkHeader* getChunkHeader() const noexcept;
+
+    /// @brief Get the pointer to the user-payload
+    /// @return the pointer to the user-payload
+    void* getUserPayload() noexcept;
+
+    /// @brief Get the const pointer to the user-payload
+    /// @return the const pointer to the user-payload
+    const void* getUserPayload() const noexcept;
+
+    friend class ServerPortUser;
+
+  protected:
+    uint8_t m_rpcHeaderVersion{RPC_HEADER_VERSION};
+    uint32_t m_lastKnownClientQueueIndex{UNKNOWN_CLIENT_QUEUE_INDEX};
+    UniquePortId m_clientQueueUniquePortId;
+    int64_t m_sequenceId{0};
+};
+
+class RequestHeader : public RpcBaseHeader
+{
+  public:
+    /// @brief Constructs and initializes a RpcBaseHeader
+    /// @param[in] clientQueueUniquePortId is the UniquePortId of the port owning the client queue where the response
+    /// shall be delivered
+    /// @param[in] lastKnownClientQueueIndex is the last know index of the client queue in the ChunkDistributor for fast
+    /// lookup
+    explicit RequestHeader(const UniquePortId& clientQueueUniquePortId,
+                           const uint32_t lastKnownClientQueueIndex) noexcept;
+
+    RequestHeader(const RequestHeader& other) = delete;
+    RequestHeader& operator=(const RequestHeader&) = delete;
+    RequestHeader(RequestHeader&& rhs) = default;
+    RequestHeader& operator=(RequestHeader&& rhs) = default;
+    ~RequestHeader() = default;
+
+    /// @brief Sets the sequence ID which is used to match a response to a request
+    /// @param[in] sequenceId is a consecutive number set by the user
+    /// @note The user has to set this manually if multiple requests are sent before a response is read since a server
+    /// might drop a requests or process the requests out of order and therefore the responses might also be out of
+    /// order
+    void setSequenceId(const int64_t sequenceId) noexcept;
+
+    /// @brief Sets the fire and forget flag which indicates that no response is expected
+    void setFireAndForget() noexcept;
+
+    /// @brief Obtains the fire and forget flag
+    /// @return true if set and no response is expected, false otherwise
+    bool getFireAndForget() const noexcept;
```

Comment:
Lyrical speaking I would prefer the method name `isFireAndForget` this reads also much better in an if statement. Otherwise you may wonder what kind of type it returns when you read `if ( getFireAndForget() )` but with `is` or `has` it is for me always a bool